src/lib/AudioUltra/Controls/Html5Player.ts

```diff
@@ -0,0 +1,82 @@
+import { WaveformAudio } from '../Media/WaveformAudio';
+import { Player } from './Player';
+
+export class Html5Player extends Player {
+  init(audio: WaveformAudio) {
+    super.init(audio);
+
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.on('resetSource', this.handleResetSource);
+
+    this.audio.el.addEventListener('play', this.handlePlayed);
+    this.audio.el.addEventListener('pause', this.handlePaused);
+  }
+
+  destroy() {
+    super.destroy();
+
+    if (this.audio?.el) {
+      this.audio.el.removeEventListener('play', this.handlePlayed);
+      this.audio.el.removeEventListener('pause', this.handlePaused);
+    }
+  }
+
+  protected playAudio(start?: number, duration?: number): void {
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.el.currentTime = this.currentTime;
+    this.audio.el.addEventListener('ended', this.handleEnded);
+    this.bufferPromise = new Promise(resolve => {
+      this.bufferResolve = resolve;
+    });
+
+    const time = this.currentTime;
+
+    this.audio.el
+      .play()
+      .then(() => this.bufferPromise!.then())
+      .then(() => {
+        this.timestamp = performance.now();
+
+        // We need to compensate for the time it took to load the buffer
+        // otherwise the audio will be out of sync of the timer we use to
+        // render updates
+        if (this.audio?.el) {
```

Comment:
can it be false at this moment?

Reply:
The case that comes to mind is hitting play on a large file, and it hits an unbuffered chunk, then the user clicks away on QuickView.